### PR TITLE
Fix REPL shadowing bug

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala [hunk 1]
```diff
@@ -1060,7 +1060,7 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
     if (sym.isImport)
       sym.infoOrCompleter match {
         case info: Namer#Completer => return info.original.show
-        case info: ImportType => return s"import $info.expr.show"
+        case info: ImportType => return s"import ${info.expr.show}"
```
Comment (som-snytt) [inline compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala:1063]: I sense a useful lint for a third-party linter!

File: compiler/test/dotty/tools/repl/ShadowingTests.scala [hunk 2]
```diff
@@ -97,7 +109,7 @@ class ShadowingTests extends ReplTest(options = ShadowingTests.options):
          |""".stripMargin
   )
 
-  @Test def `shadow subdirectories on classpath` =
+  @Test def shadowSubDir =
```
Comment (som-snytt) [inline compiler/test/dotty/tools/repl/ShadowingTests.scala:112]: I kind of understand the first form.

**Comment by som-snytt** [comment]
I've only a minute between grocery shopping and chauffeuring to piano lesson. I didn't grok in a minute. It was about 0.5 grok/min.

I recognize checking for repl wrapper aka Wrapl.

**Comment by som-snytt** [comment]
oh wait, just realized I got someone fired. Unless they were already fired. Or quit.

File: compiler/test/dotty/tools/repl/ShadowingTests.scala (hunk 2 above)
Comment (dwijnand) [reply compiler/test/dotty/tools/repl/ShadowingTests.scala:112]: I didn't want to try to figure out how to glob for spaces.  After testing it now I can't find a way, I can only star it like `-- *shadow*sub*`...

File: compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala (hunk 1 above)
Comment (mpollmeier) [reply compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala:1063]: 🤯 wow, seeing this in dotc... I had to check: this dates back to 2016

Out of curiosity I started my custom "third-party linter"
```
git grep 's".*\$[a-zA-Z0-9_-]*\..*"'|grep -v '{'
```
and didn't find any other obvious cases like this one. 

Going through those does demonstrates however that it'd be quite tricky to judge automatically if it's a bug or not... e.g. consider these:
```
community-build/src/scala/dotty/communitybuild/projects.scala:  override val testCommand = s"$baseCommand.test"
compiler/src/dotty/tools/dotc/printing/PlainPrinter.scala:    s"import $exprStr.$selectorStr"
compiler/src/dotty/tools/dotc/util/Spans.scala:    //assert(start <= end || start == 1 && end == 0, s"$start..$end")
compiler/src/dotty/tools/runner/ScalaClassLoader.scala:      throw new NoSuchMethodException(s"$objectName.main is not static")
```

**Comment by dwijnand** [comment]
Excellent, thank you.  I had the `io` test in `ShadowingTests` as the failing reproduction, but it's good to check manually too.

**Review by som-snytt — APPROVED** [review]

**Pull request merged**